File: PythonNBA/scripts/_check_hist.py
# scripts/_check_hist.py
import json

def main():
    with open("data/history.json", "r") as f:
        h = json.load(f)
    runs = h.get("runs", [])
    print("Total runs:", len(runs))
    print("First:", runs[0].get("dateDisplay") if runs else None,
          "Last:", runs[-1].get("dateDisplay") if runs else None)
    print("Burn-in runs:", sum(1 for r in runs if r.get("burnIn")))
    print("Live runs:", sum(1 for r in runs if not r.get("burnIn")))

    # Count graded picks
    w = 0
    l = 0
    for r in runs:
        if r.get("burnIn"):
            continue
        for g in r.get("games", []):
            if g.get("sResult") == "WIN":
                w += 1
            elif g.get("sResult") == "LOSS":
                l += 1
    total = w + l
    pct = f"{w / total * 100:.1f}" if total > 0 else "0.0"
    print(f"Spread record: {w}-{l} ({pct}%)")
    print("Stored weights:", json.dumps(h.get("weights", {}))[:200])
    print("Stored residualVar:", h.get("residualVar"))

if __name__ == "__main__":
    main()

File: jsNBA/scripts/_check_hist.mjs
import fs from 'fs';
const h = JSON.parse(fs.readFileSync('data/history.json','utf8'));
const runs = h.runs || [];
console.log('Total runs:', runs.length);
console.log('First:', runs[0]?.dateDisplay, 'Last:', runs[runs.length-1]?.dateDisplay);
console.log('Burn-in runs:', runs.filter(r => r.burnIn).length);
console.log('Live runs:', runs.filter(r => !r.burnIn).length);

// Count graded picks
let w = 0, l = 0;
for (const r of runs) {
  if (r.burnIn) continue;
  for (const g of (r.games || [])) {
    if (g.sResult === 'WIN') w++;
    else if (g.sResult === 'LOSS') l++;
  }
}
console.log(`Spread record: ${w}-${l} (${(w/(w+l)*100).toFixed(1)}%)`);
console.log('Stored weights:', JSON.stringify(h.weights || {}).slice(0, 200));
console.log('Stored residualVar:', h.residualVar);
